### ISSUE TITLE
Data loading pipeline

### ISSUE THREAD

**Issue by xeTaiz** [xeTaiz/torchvtk#6]
Rough outline of volume loading pipeline:
1. Build preprocessor for specific datasets using existing data loading utilities. The result of preprocessing is a canonical representation using a normalized `torch.Tensor` of shape `[C, D, H, W]` (Can we benefit from [channels last tensor layout](https://pytorch.org/tutorials/intermediate/memory_format_tutorial.html)?) If multiple volumes are loaded this way, enable grouping (e.g. `(volume, mask)`). Options:
    - Have datasets for `volumes` and `masks` separately, merge in data loader
    - Enforce grouping in canonical form as dictionary?
    - Enable also mixing the above 2 options
2. Serialize the most efficient way, using as small `dtype` as possible and with the best strategy. The best strategy will have to be determined using benchmarks
3. Provide dataset compatible with serialized canonical data that let's you use subsets and (possibly cached optional?) resized version. 
4. Aggregate batches using dataloaders that can.
5. Apply data augmentation. Should be mostly on GPU and process batches instead of individual volumes whenever possible.

Please add ideas and discuss
